pkg/core/native/native_test/neo_test.go

```diff
@@ -118,6 +118,47 @@ func TestNEO_CandidateEvents(t *testing.T) {
 	require.Equal(t, 0, len(aer.Events))
 }
 
+func TestNEO_VoteForNull(t *testing.T) {
+	neoCommitteeInvoker := newNeoCommitteeClient(t, 100_0000_0000)
+	neoValidatorsInvoker := neoCommitteeInvoker.WithSigners(neoCommitteeInvoker.Validator)
+	e := neoCommitteeInvoker.Executor
+	voter := e.NewAccount(t, 100_0000_0000)
+	candidate := e.NewAccount(t)
+	cfg := e.Chain.GetConfig()
+	committeeSize := cfg.GetCommitteeSize(0)
+	validatorsCount := cfg.GetNumOfCNs(0)
+	freq := validatorsCount + committeeSize
+	getAccountState := func(t *testing.T, account util.Uint160) *state.NEOBalance {
+		stack, err := neoCommitteeInvoker.TestInvoke(t, "getAccountState", account)
+		require.NoError(t, err)
+		as := new(state.NEOBalance)
+		err = as.FromStackItem(stack.Pop().Item())
+		require.NoError(t, err)
+		return as
+	}
+	advanceChain := func(t *testing.T) {
+		for i := 0; i < freq; i++ {
+			neoCommitteeInvoker.AddNewBlock(t)
+		}
+	}
+	txes := make([]*transaction.Transaction, 0, committeeSize*4-2)
+	transferTx := neoValidatorsInvoker.PrepareInvoke(t, "transfer", e.Validator.ScriptHash(), candidate.(neotest.SingleSigner).Account().PrivateKey().GetScriptHash(), int64(committeeSize+1)*1000000, nil)
+	txes = append(txes, transferTx)
+	registerTx := neoValidatorsInvoker.WithSigners(candidate).PrepareInvoke(t, "registerCandidate", candidate.(neotest.SingleSigner).Account().PublicKey().Bytes())
+	txes = append(txes, registerTx)
+	voteTx := neoValidatorsInvoker.WithSigners(voter).PrepareInvoke(t, "vote", voter.(neotest.SingleSigner).Account().PrivateKey().GetScriptHash(), nil)
```

Comment:
Should firstly vote for a candidate, then check the expected account state, then vote for nil.